### DISCUSSION TITLE
Internal timekeeping - a topic for discussion

### DISCUSSION THREAD

**Discussion by okms** [Puzzlepart/did#870] [Feature request]
[Original discussion in Teams in Norwegian](https://teams.microsoft.com/l/message/19:876c3ec2e97e4186b71cbd01f5002963@thread.tacv2/1606226828478?tenantId=dcb89c10-67b8-48cc-8819-85de37e8af91&groupId=46adb6c8-8c09-409f-a333-2d131ab6aff8&parentMessageId=1606226828478&teamName=Did&channelName=General&createdTime=1606226828478)

"Customer key" and "Name" are just display titles for CustomerKey and ProjectKey.   
Maybe we should have their display name as a configurable setting?  
That way a tenant could choose to have e.g.
`[orgname] [projectname]`  
`[project] [account]`  
`[region][country]`
`[county][project]`  
or whatever other 2-level structure they'd want

Additional properties could be supported via an AdditionalFields column in cosmos. This field would contain a JSON blob with any additonal metadata per entitiy.

We would need

UI for adding / removing addional fields per entity (Customer / Project / User )
Expose additional props in reports
Expose in TimeSheet?

**Comment by okms** [comment]
Kommentarer 

![image](https://user-images.githubusercontent.com/170147/100210214-08941400-2f0b-11eb-9748-4685c7732701.png)

**Comment by damsleth** [comment]
"Customer key" and "Name" are just display titles for CustomerKey and ProjectKey.   
Maybe we should have their display name as a configurable setting?  
That way a tenant could choose to have e.g.
`[orgname] [projectname]`  
`[project] [account]`  
`[region][country]`
`[county][project]`  
or whatever other 2-level structure they'd want

**Comment by okms** [comment]
Additional properties could be supported via an `AdditionalFields` column in table store / cosmos. This field would contain a JSON blob with any additonal metadata per entitiy.

We would need

- UI for adding / removing addional fields per entity (Customer / Project / User )
- Expose additional props in reports
- Expose in TimeSheet?

**Comment by olemp** [comment]
@okms @damsleth Maybe it would make sense to use Account and Delivery like in Dynamics instead of Customer and Project?
